File: dataset/human/c/file_520.c
#include<stdio.h>
double abso(double a,double b){
  if(a-b<0)
    return b-a;
  else
    return a-b;
}
int main(){
  int n,i;
  scanf("%d",&n);
  double a[n],sum=0,x,y=0,min;
  scanf("%lf",&x);
  for(i=1;i<n;i++){
    scanf("%lf",&a[i]);
    sum+=a[i];
  }
  min=abso(x,sum);
  for(i=1;i<n-1;i++){
    x+=a[i];
    sum-=a[i];
    if(abso(x,sum)<min)
      min=abso(x,sum);
  }
  printf("%.0lf\n",min);
}

File: dataset/human/python/file_520.py
import sys
import numpy as np
input = sys.stdin.readline

def main():
	N = int(input())
	A = list(map(int,input().split()))
	A.sort()
	count = 0
	m = 10**10
	if 0 in A:
		A = np.array(A)
		print(sum(abs(A)))
	else:
		for i in range(N):
			if A[i]<0:
				count += 1
			m = min(m,abs(A[i]))
		A = np.array(A)
		if count%2 == 0:
			print(sum(abs(A)))
		else:
			print(sum(abs(A))-2*m)

if __name__ == "__main__":
	main()
